src/components/FlexBox/FlexBox.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { ReactNode, HTMLAttributes } from 'react';
+import { dcnb } from 'cnbuilder';
+import * as styles from './FlexBox.styles';
+import * as types from './FlexBox.types';
+
+type FlexBoxProps = HTMLAttributes<HTMLElement> & {
+  children?: ReactNode;
+  as?: types.FlexElementType;
+  direction?: types.FlexDirectionType;
+  wrap?: types.FlexWrapType;
+  gap?: boolean;
+  justifyContent?: types.FlexJustifyContentType;
+  alignContent?: types.FlexAlignContentType;
+  alignItems?: types.FlexAlignItemsType;
+};
+
+export const FlexBox = ({
+  as: AsComponent = 'div',
+  direction,
+  wrap,
+  gap,
+  justifyContent,
+  alignContent,
+  alignItems,
+  className,
+  children,
+  ...props
+}: FlexBoxProps) => (
+  <AsComponent
```

Comment:
General clean up of these ported components for typescript consistency and not using `React.createElement `anymore